sound/soc/sof/ipc4-topology.c

```diff
@@ -986,7 +986,7 @@ static int sof_ipc4_init_audio_fmt(struct snd_sof_dev *sdev,
 		channels = SOF_IPC4_AUDIO_FORMAT_CFG_CHANNELS_COUNT(fmt->fmt_cfg);
 		valid_bits = SOF_IPC4_AUDIO_FORMAT_CFG_V_BIT_DEPTH(fmt->fmt_cfg);
 		if (params_rate(params) == rate && params_channels(params) == channels &&
-		    sample_valid_bits == valid_bits) {
+		    sample_valid_bits == valid_bits && !pin_fmt->pin_index) {
```

Comment:
I don't prefer to mix pin_fmt->pin_index with rate or channels, they are different things. How about the following one. It is close to comments and easy to understand
````c
for (i = 0; i < num_formats && !pin_fmt->pin_index; i++, pin_fmt++) 
````

Reply:
I would rather spell this out either checking moving the pin_index check as the first or a separate test
```
if (pin_fmt[i].pin_index)
	continue;
```
The pins stored in random order?

Reply:
We only support pin0 currently? If this is the case then prior to this patch nothing works? Should this be squashed to some previous patch?

Reply:
no we only have pin0 formats in tplg today. So everything works even before this patch. But before we start adding other pin formats, we need to spell this out so that base_cfg doesn't use non-zero formats

Reply:
changed to make the pin_index check explicit now